src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import Avatar from '@material-ui/core/Avatar';
+import { Button } from '@material-ui/core';
+import './Dropdown.scss';
+
+interface Dropdownprops {
+  DropdownOpener: React.ReactNode;
+}
+
+export const Dropdown: React.FC<Dropdownprops> = ({ DropdownOpener }) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState<boolean>(false);
+  const box = useRef(null);
+
+  const closeDropdownOutside = (ref: React.MutableRefObject<HTMLDivElement | null>) => {
+    useEffect(() => {
```

Comment:
I would move this useEffect outside the function.

Reply:
I agree

Reply:
If I move the useEffect outside, I won't be able to use ref as dependencies in useEffect [isDropdownOpen, ref] 
and the functionality will stop working. 
It's a functionality from the internet and I'm not 100 percent sure how it works : (, 
so it's gonna be okay if i leave it the way it is?

Reply:
ok! :)

Reply:
You can pass ref as prop to this hook like I did
```
export const useIsContentOverflowing = (ref: MutableRefObject<HTMLElement>) => {
  const [isContentOverflowing, setIsContentOverflowing] = useState(false);

  useEffect(() => {
    const handleResize = () => {
      setIsContentOverflowing(ref.current.offsetHeight < ref.current.scrollHeight);
    };
    window.addEventListener('resize', handleResize);
    handleResize();
    return () => window.removeEventListener('resize', handleResize);
  }, []);
  return isContentOverflowing;
};
```